S/Singular/build_tarballs.jl

```diff
@@ -1,13 +1,38 @@
 # Note that this script can accept some limited command-line arguments, run
 # `julia build_tarballs.jl --help` to see a usage message.
 using BinaryBuilder, Pkg
+import Pkg.Types: VersionSpec
 
+# Singular_jll versions are decoupled from the upstream versions.
+# Whenever we package a new official Singular release, we initially map its
+# version X.Y.Z to X00.Y00.Z00 (i.e., multiply each component by 100).
+# So for example version 4.1.3 would become 400.100.300.
+# For sub-patch level release such as 4.1.3p5, one has to decide based on the actual
+# changes how to map this to a JLL version, as Singular releases do not follow semver.
+#
+# Moreover, all our packages using Singular_jll use `~` in their compat ranges.
+#
+# Together, this allows us to increment the patch level of the JLL for minor tweaks.
+# If a rebuild of the JLL is needed which keeps the upstream version identical
+# but breaks ABI compatibility for any reason, we can increment the minor version
+# e.g. go from 400.100.300 to 400.101.300.
+# To package prerelease versions, we can also adjust the minor version; e.g. we may
+# map a prerelease of 4.1.4 to 400.100.390 or 400.101.390 (depending on whether it
+# contains ABI changes).
+#
+# There is currently no plan to change the major version, except when Singular itself
+# changes its major version. It simply seemed sensible to apply the same transformation
+# to all components.
+#
+# WARNING WARNING WARNING: any change to the the version of this JLL should be carefully
+# coordinated with corresponding changes to Singular_jll.jl, LoadFlint.jl, Nemo.jl,
+# and possibly other packages.
 name = "Singular"
-version = v"4.1.4"  # this is actually a snapshot made after 4.1.3p5
+version = v"400.101.390"  # a snapshot made after 4.1.3p5, but with major new features (next version might be 4.2.0?)
```

Comment:
We just talked to Hans and I learned that the version scheme is as follows: `4.1` stays until the Singular language itself is changed -- which may be next week or never, but in any case, is a major change. Therefore I think we should merge these together into the *major* version of the JLL. Then the `3p5` in `4.1.3p5` really correspond to what we'd normally call "minor" and "patch level".

All together, I suggest that we map `4.1.3p5` to `401.300.500`  ; the next Singular release (planned to be released before Christmas) will be `4.1.4` and will probably be followed by `4.1.4p1`, `4.1.4p2`, etc.). So we can label our snapshots e.g. `401.390.000`  and that fits again with the usual scheme.

I'll adjust this PR to match right now, but please holler if you think I am nuts (or if you have any less offensive feedback, too ;-))